### DISCUSSION TITLE
question: How can I validate a union of strings?

### DISCUSSION THREAD

**Discussion by stefan-bc-ts** [typestack/class-validator#2362] [Questions & Answers]
**I was trying to...**
use ```export type ValidateValues = ‘val1’ | ‘val2’ | ‘val3’;```
with @Type to make sure the values are correct

**The problem:**

```
@Type((_) => ValidateValues)
'ValidateValues' only refers to a type, but is being used as a value here
```

PS: I was trying to avoid using a enum

**Comment by braaar** [comment]
I'm moving this to the discussion tab, as we are trying to keep the issue count low. Please see [this issue](https://github.com/typestack/class-validator/issues/1775) for an explanation.

**Comment by jessekernaghan** [comment]
For anyone coming across this, I have been solving this by declaring an array `as const` and then using it with `@IsIn()`, i.e.:

```typescript
const validateValues = ['val1', 'val2', 'val3'] as const; // note the 'as const'
type ValidateValues = (typeof validateValues)[number]; // produces 'val1' | 'val2' | 'val3'

class MyClass {
  @IsIn(validateValues)
  type: ValidateValues;
}
```
